File: kotlinExamples/src/test/java/com/example/EmailAddressTests.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

public class EmailAddressTests {

  @Test
  public void parsing_java() {
    assertEquals(
        new EmailAddress("fred", "example.com"),
        EmailAddress.parse("fred@example.com")
    );
  }

  @Test
  public void parsing_failure_java() {
    assertThrows(
        IllegalArgumentException.class,
        () -> EmailAddress.parse("@")
    );

  }
}

File: kotlinExamples/src/test/kotlin/gradle/project/kotlinExamples/EmailAddressTests.kt
package gradle.project.kotlinExamples

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

class EmailAddressTests {

    @Test
    fun parsing_kotlin() {
        val expected = EmailAddress("myname", "domain.com")
        assertEquals(
            EmailAddress.parse("myname@domain.com"),
            expected
        )
    }

    @Test
    fun parsing_failure_kotlin() {
        assertFailsWith<IllegalArgumentException> {
            EmailAddress.parse("@")
        }
    }
}